File: src/main/webapp/athome/src/app/rpi-backlight/rpi-backlight.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {RpiBackLight} from "./rpi-backlight.model";
import {Observable} from "rxjs";

@Injectable()
export class RpiBackLightService {
  url = '/api/rpi-backlight';

  constructor(private httpClient: HttpClient) {

  }

  getBackLightIntensity(): Observable<number> {
    return this.httpClient.get<number>(this.url);
  }

  setBackLightIntensity(rpiBackLight:RpiBackLight): Observable<RpiBackLight> {
    return this.httpClient.post<RpiBackLight>(this.url, rpiBackLight);
  }
}

File: src/main/java/org/jorgma/athome/rpi/service/RpiBackLightService.java
package org.jorgma.athome.rpi.service;

import org.jorgma.athome.rpi.dao.RpiBackLight;

import java.io.IOException;

/**
 * Created by jorgma on 2018-03-24.
 */
public interface RpiBackLightService {
    int getBackLightIntensity();

    void setBackLightIntensity(RpiBackLight rpiBackLight) throws IOException;
}
